File: HW3/src/main/java/refactoring/utils/DatabaseUtils.java
package refactoring.utils;

import org.jetbrains.annotations.NotNull;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

public class DatabaseUtils {
    private final @NotNull Connection connection;
    private final @NotNull Statement statement;

    public DatabaseUtils(@NotNull String url, @NotNull String user, @NotNull String password) throws SQLException {
        connection = DriverManager.getConnection(url, user, password);
        statement = connection.createStatement();
    }

    public void close() throws SQLException {
        statement.close();
        connection.close();
    }

    public @NotNull Statement getStatement() {
        return statement;
    }
}

File: S7/HW4/src/main/kotlin/easy/soc/hacks/hw4/utils/DatabaseUtils.kt
package easy.soc.hacks.hw4.utils

import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException
import java.sql.Statement

object DatabaseUtils {
    private val connection: Connection = DriverManager.getConnection("jdbc:sqlite:todo.db", "", "")
    val statement: Statement = connection.createStatement()

    @Throws(SQLException::class)
    fun close() {
        statement.close()
        connection.close()
    }
}